caikit/interfaces/nlp/data_model/classification.py

```diff
@@ -0,0 +1,136 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Data structures for classification representations"""
+
+# Standard
+from typing import List, Optional
+
+# Third Party
+import numpy as np
+
+# First Party
+from py_to_proto.dataclass_to_proto import Annotated, FieldNumber
+import alog
+
+# Local
+from ....core import DataObjectBase, dataobject
+from .package import NLP_PACKAGE
+from .text_generation import FinishReason
+
+log = alog.use_channel("DATAM")
+
+
+@dataobject(package=NLP_PACKAGE)
+class ClassificationTrainRecord(DataObjectBase):
+    """A classification training record consisting of a single train instance."""
+
+    text: Annotated[str, FieldNumber(1)]  # Text to be classified
+    labels: Annotated[
+        List[str], FieldNumber(2)
+    ]  # Class labels to be learnt for the text
+
+
+@dataobject(package=NLP_PACKAGE)
+class Classification(DataObjectBase):
```

Comment:
nit: I think `classification` usually refers to the "task name", so having the same name as internals of DM can be a bit confusing. Do we want to name this one as `ClassificationResult` or `ClassificationPrediction` ?